web/modules/custom/weather/weather.install

```diff
@@ -35,19 +35,16 @@ function weather_schema() {
     'primary key' => ['id'],
   ];
   return $schema;
+}
 
-  /**
-   * Implements hook_update_N().
-   */
-  function weather_update_10001(): void {
-    if (\Drupal::service('database.schema')->tableExists('weather_info')) {
-      return;
-    }
-
+/**
+ * Implements hook_update_N().
+ */
+function weather_update_10001(): void {
+  if (\Drupal::service('database')->schema()->tableExists('weather_info')) {
     $schema = weather_schema();
     $table = $schema['weather_info'];
 
-    \Drupal::database()->schema()->createTable('weather_info', $table);
+    \Drupal::database()->schema()->dropTable('weather_info', $table);
```

Comment:
"Weather" PR was already accepted, so we could assume this module was installed on the production server and update_10001 was already run.

So that update function will not run again, and the table will not be dropped.
You should create a new update function for dropping the table.

The schema function is obsolete now as well.